scripts/consensus_merge.py

```diff
@@ -0,0 +1,636 @@
+    alt_depth = int(np.mean([int(f.split(',')[1]) for f in ad_list if f != '.']))
+    return (ref_depth, alt_depth)
+
+def get_af_consensus(af_list):
+    """ Get mean value of AF across callers
+        Args:
+            af_list (list): list of AF (strings)
+        Return:
+            (float): mean AF
+    """
+    return np.mean([float(f) for f in af_list if f != '.'])
+
+def get_dp_consensus(dp_list):
+    """ Get mean value of DP across callers
+        Args:
+            dp_list (list): list of DP (strings)
+        Return:
+            (int): mean DP
+    """
+    return int(np.mean([int(f) for f in dp_list if f != '.']))
+
+def get_mapq(cram, chrom, pos):
+    """ Get RMS mapping quality and number of MAPQ=0 reads at locus
+
+        Args:
+            cram (pysam.AlignmentFile): open in 'r' mode
+            chrom (str): locus chromosome name
+            pos (int): 1-based locus coordinate
+
+        Returns:
+            tuple of ints RMS mapq and # of mapq reads at locus
+    """
+    mapq = []
+    mq0 = 0
+#    print(chrom, pos-1, pos)
+
+    aligned_reads = cram.fetch(chrom, pos-1, pos)
+
+    for read in aligned_reads:
+        mapq.append(read.mapq)
+        if read.mapq == 0:
+            mq0 += 1
+
+    if not mapq:
+        return 0, 0
+
+    rms_mapq = np.sqrt(np.mean(np.array(mapq)**2))
+    return int(rms_mapq), mq0
+
+def build_output_record(single_caller_variants, output_vcf, normal_cram, hotspot=False):
+    """ Create a single VCF record for the consensus output
+            by interrogating the single-caller records for that variant
+            and add that record the consensus VCF
+
+        Args: 
+            single_caller_variants (list of Variants): list of Variants
+                representing a single call in one or more callers
+            output_vcf (pysam.VariantFile): consensus file, open in 'w' mode
+            normal_cram (pysam.Alignmentfile): Normal CRAM for this sample,
+                open in 'r' mode
+            hotspot (bool): whether this variant is in a mutational hotspot
+                (default False)
+    """
+    variant_lookup = {v.caller: v for v in single_caller_variants}
+
+    output_record = output_vcf.new_record()
+    # Set consistent attributes
+    liftover_record = single_caller_variants[0].record
+    output_record.chrom = liftover_record.chrom
+    output_record.ref = liftover_record.ref
+    output_record.alts = liftover_record.alts
+    output_record.id = liftover_record.id
+    output_record.start = liftover_record.start
+    output_record.stop = liftover_record.stop
+
+    # For each caller, get information required for format fields for this variant
+    joint_tags = ('GT', 'AD', 'AF', 'DP')
+    no_call_info = {tag: '.' for tag in joint_tags} 
+    allele_information = {caller.lower(): {} for caller in CALLER_NAMES}
+
+    for variant in single_caller_variants:
+        normal_sample, tumor_sample = variant.record.samples
```

Comment:
This is potentially a problem. I was assuming that the first sample was always the normal based on inspecting the single-caller vcfs. If this is not reliably true, probably your solution of passing in the names is the way to go. Is it possible to get any certainty on this, or had we better not assume?

Reply:
Although I should say that it is *only* potentially a problem because the means of deriving a "normal" genotype in Strelka2 is different for tumor and normal samples. If, as we were discussing, the standardization of the Strelka2 VCF is moving upstream of this piece, there is no need to get this right, as the normal and tumor samples are otherwise handled the same and indeed the whole tumor/normal logical aspect of this will be going away.

Reply:
Ah, yeah, that's a good point. In that case, we'd add that input upstream and not worry about it here, as long as you use the sample ID as the key in the off-chance one of the vcfs has them reversed, otherwise you might be putting tumor counts for a single caller in the normal field and vice-versa without knowing. Does that make sense?

Reply:
Yes, it keeps track of samples by ID rather than assuming a particular ordering.